plan/expression_rewriter.go

```diff
@@ -764,7 +765,11 @@ func (er *expressionRewriter) rewriteVariable(v *ast.VariableExpr) {
 		er.err = errors.Trace(err)
 		return
 	}
-	er.ctxStack = append(er.ctxStack, datumToConstant(types.NewStringDatum(val), mysql.TypeString))
+	e := datumToConstant(types.NewStringDatum(val), mysql.TypeString)
```

Comment:
`e := datumToConstant(types.NewStringDatum(val), mysql.TypeVarString)`, line 769 can be removed

Reply:
done